src/Microsoft.OData.Core/JsonLight/ODataJsonLightErrorDeserializer.cs

```diff
@@ -333,8 +333,15 @@ private void ReadPropertyValueInInnerError(int recursionDepth, ODataInnerError i
                     break;
 
                 default:
-                    // skip any unsupported properties in the inner error
-                    this.JsonReader.SkipValue();
+                    if (!innerError.Properties.ContainsKey(propertyName))
+                    {
+                        innerError.Properties.Add(propertyName, this.JsonReader.ReadODataValue());
+                    }
+                    else
+                    {
+                        innerError.Properties[propertyName] = this.JsonReader.ReadODataValue();

```

Comment:
Yes. This may be undesired but to keep the behavior same, this was the only option. For instance, a json payload with duplicate properties would not throw error prior to my change. I don't want to throw an error now. 

---
In reply to: [323950924](https://github.com/OData/odata.net/pull/1523#discussion_r323950924) [](ancestors = 323950924)